src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,63 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
     public void getStatistic(String fromFileName, String toFileName) {
+        File inputCsvFile = new File(fromFileName);
+        int totalSupplyAmount = calculateTotalValue(inputCsvFile, "supply");
+        int totalBuyAmount = calculateTotalValue(inputCsvFile, "buy");
+        writeReport(toFileName, totalSupplyAmount, totalBuyAmount);
+    }
```

Comment:
The `getStatistic` method is well-structured and adheres to the principle that methods should be responsible for one task. However, consider reducing visibility of the method if it's not intended to be used outside the package, or if it's not part of a public API.